members-only/app/models/user.rb

```diff
@@ -0,0 +1,46 @@
+class User < ApplicationRecord
+	attr_accessor :remember_token
+	before_save { self.email = email.downcase }
+	before_create :remember_token
+	has_many :post, dependent: :destroy
+
+  validates :name,  presence: true, length: { maximum: 50 }
+  VALID_EMAIL_REGEX = /\A[\w+\-.]+@[a-z\d\-.]+\.[a-z]+\z/i
+  validates :email, presence: true, length: { maximum: 255 },
+                    format: { with: VALID_EMAIL_REGEX },
+                    uniqueness: { case_sensitive: false }
+  has_secure_password
+  validates :password, presence: true, length: { minimum: 6 }
+
+  # Returns the hash digest of the given string.
+  def User.digest(string)
+    cost = ActiveModel::SecurePassword.min_cost ? BCrypt::Engine::MIN_COST :
+                                                  BCrypt::Engine.cost
+    BCrypt::Password.create(string, cost: cost)
+  end
```

Comment:
Please, read that https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/authentication/before_create_requirement.png